cmd/keeper/keeper.go

```diff
@@ -1512,7 +1513,8 @@ func (p *PostgresKeeper) postgresKeeperSM(pctx context.Context) {
 
 	if !reflect.DeepEqual(db.Spec.PGHBA, pgm.CurHba()) {
 		log.Infow("postgres hba entries changed, reloading postgres instance")
-		pgm.SetHba(db.Spec.PGHBA)
+		// Dynamicly generate hba auth from clusterData
+		pgm.SetHba(append(db.Spec.PGHBA, p.generateHBA(cd)...))
```

Comment:
The generated hba entries should be inserted before the user provided hba entries.